File: solutions/3074. Apple Redistribution into Boxes/3074.cpp
// Time complexity: O(n+mlogm)
// Space complexity: O(1)
class Solution {
public:
  int minimumBoxes(const vector<int>& apple, vector<int>& capacity) {
    int sum = accumulate(apple.begin(), apple.end(), 0);
    int i = 0;
    sort(capacity.begin(), capacity.end(), greater<int>());
        
    for ( ; i < capacity.size() && sum > 0; ++i) 
      sum -= capacity[i];

    return i;
  }
};

File: solutions/3074. Apple Redistribution into Boxes/3074.py
# Time complexity: O(n+mlogm)
# Space complexity: O(1)
class Solution:
  def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
    sumApple = sum(apple)
    i = 0
    capacity.sort(reverse=True)

    while i < len(capacity) and sumApple > 0:
      sumApple -= capacity[i]
      i += 1

    return i